<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Evolutionary models | An extra gentle introduction to molecular evolution</title>
  <meta name="description" content="5 Evolutionary models | An extra gentle introduction to molecular evolution" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Evolutionary models | An extra gentle introduction to molecular evolution" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="sjspielman/gentle_intro_molecular_evolution" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Evolutionary models | An extra gentle introduction to molecular evolution" />
  
  
  

<meta name="author" content="Stephanie J. Spielman" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="mutiple-sequence-alignments.html"/>
<link rel="next" href="building-your-phylogeny.html"/>
<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />












</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="background-molecular-evolution.html"><a href="background-molecular-evolution.html"><i class="fa fa-check"></i><b>2</b> Background: Molecular Evolution</a>
<ul>
<li class="chapter" data-level="2.1" data-path="background-molecular-evolution.html"><a href="background-molecular-evolution.html#evolution-in-populations"><i class="fa fa-check"></i><b>2.1</b> Evolution in populations</a></li>
<li class="chapter" data-level="2.2" data-path="background-molecular-evolution.html"><a href="background-molecular-evolution.html#evolution-in-dna"><i class="fa fa-check"></i><b>2.2</b> Evolution in DNA</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="background-molecular-evolution.html"><a href="background-molecular-evolution.html#a-brief-digression-snps"><i class="fa fa-check"></i><b>2.2.1</b> A brief digression: SNPs</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="background-molecular-evolution.html"><a href="background-molecular-evolution.html#fitness-consequences-of-mutations"><i class="fa fa-check"></i><b>2.3</b> Fitness consequences of mutations</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="background-interpreting-building-and-using-phylogenies.html"><a href="background-interpreting-building-and-using-phylogenies.html"><i class="fa fa-check"></i><b>3</b> Background: Interpreting, building, and using phylogenies</a>
<ul>
<li class="chapter" data-level="3.1" data-path="background-interpreting-building-and-using-phylogenies.html"><a href="background-interpreting-building-and-using-phylogenies.html#reading-and-interpreting-phylogenies"><i class="fa fa-check"></i><b>3.1</b> Reading and interpreting phylogenies</a></li>
<li class="chapter" data-level="3.2" data-path="background-interpreting-building-and-using-phylogenies.html"><a href="background-interpreting-building-and-using-phylogenies.html#the-pipeline-of-phylogenetic-reconstruction"><i class="fa fa-check"></i><b>3.2</b> The pipeline of phylogenetic reconstruction</a></li>
<li class="chapter" data-level="3.3" data-path="background-interpreting-building-and-using-phylogenies.html"><a href="background-interpreting-building-and-using-phylogenies.html#analyzing-sequences-in-a-phylogenetic-context"><i class="fa fa-check"></i><b>3.3</b> Analyzing sequences in a phylogenetic context</a></li>
<li class="chapter" data-level="3.4" data-path="background-interpreting-building-and-using-phylogenies.html"><a href="background-interpreting-building-and-using-phylogenies.html#gene-tree-vs-species-trees"><i class="fa fa-check"></i><b>3.4</b> Gene tree vs species trees</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="mutiple-sequence-alignments.html"><a href="mutiple-sequence-alignments.html"><i class="fa fa-check"></i><b>4</b> Mutiple sequence alignments</a></li>
<li class="chapter" data-level="5" data-path="evolutionary-models.html"><a href="evolutionary-models.html"><i class="fa fa-check"></i><b>5</b> Evolutionary models</a>
<ul>
<li class="chapter" data-level="5.1" data-path="evolutionary-models.html"><a href="evolutionary-models.html#background-why-use-evolutionary-models"><i class="fa fa-check"></i><b>5.1</b> Background: Why use evolutionary models?</a></li>
<li class="chapter" data-level="5.2" data-path="evolutionary-models.html"><a href="evolutionary-models.html#markov-models"><i class="fa fa-check"></i><b>5.2</b> Markov models</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="evolutionary-models.html"><a href="evolutionary-models.html#key-properties-of-markov-processes"><i class="fa fa-check"></i><b>5.2.1</b> Key properties of Markov processes</a></li>
<li class="chapter" data-level="5.2.2" data-path="evolutionary-models.html"><a href="evolutionary-models.html#markov-chains-and-multiple-sequence-alignments"><i class="fa fa-check"></i><b>5.2.2</b> Markov chains and multiple sequence alignments</a></li>
<li class="chapter" data-level="5.2.3" data-path="evolutionary-models.html"><a href="evolutionary-models.html#protein-models"><i class="fa fa-check"></i><b>5.2.3</b> Protein models</a></li>
<li class="chapter" data-level="5.2.4" data-path="evolutionary-models.html"><a href="evolutionary-models.html#rate-variation-or-heterogeneity"><i class="fa fa-check"></i><b>5.2.4</b> Rate variation or heterogeneity</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="evolutionary-models.html"><a href="evolutionary-models.html#choosing-a-model"><i class="fa fa-check"></i><b>5.3</b> Choosing a model</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="building-your-phylogeny.html"><a href="building-your-phylogeny.html"><i class="fa fa-check"></i><b>6</b> Building your phylogeny</a></li>
<li class="chapter" data-level="7" data-path="assessing-confidence-in-phylogenetic-inferences.html"><a href="assessing-confidence-in-phylogenetic-inferences.html"><i class="fa fa-check"></i><b>7</b> Assessing confidence in phylogenetic inferences</a>
<ul>
<li class="chapter" data-level="7.1" data-path="assessing-confidence-in-phylogenetic-inferences.html"><a href="assessing-confidence-in-phylogenetic-inferences.html#bootstrap-and-friends"><i class="fa fa-check"></i><b>7.1</b> Bootstrap and friends</a></li>
<li class="chapter" data-level="7.2" data-path="assessing-confidence-in-phylogenetic-inferences.html"><a href="assessing-confidence-in-phylogenetic-inferences.html#pp"><i class="fa fa-check"></i><b>7.2</b> PP</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="measuring-evolutionary-rates.html"><a href="measuring-evolutionary-rates.html"><i class="fa fa-check"></i><b>8</b> Measuring evolutionary rates</a>
<ul>
<li class="chapter" data-level="8.1" data-path="measuring-evolutionary-rates.html"><a href="measuring-evolutionary-rates.html#nucleotide-and-protein-relative-rates"><i class="fa fa-check"></i><b>8.1</b> Nucleotide and protein relative rates</a></li>
<li class="chapter" data-level="8.2" data-path="measuring-evolutionary-rates.html"><a href="measuring-evolutionary-rates.html#codon-dnds"><i class="fa fa-check"></i><b>8.2</b> Codon <span class="math inline">\(dN/dS\)</span></a></li>
<li class="chapter" data-level="8.3" data-path="measuring-evolutionary-rates.html"><a href="measuring-evolutionary-rates.html#mutation-selection-models"><i class="fa fa-check"></i><b>8.3</b> Mutation-selection models</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="the-molecular-clock.html"><a href="the-molecular-clock.html"><i class="fa fa-check"></i><b>9</b> The Molecular Clock</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">An extra gentle introduction to molecular evolution</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="evolutionary-models" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Evolutionary models</h1>
<div id="background-why-use-evolutionary-models" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Background: Why use evolutionary models?</h2>
<p>Imagine this simple example of two homologous sequences. This is shown in “FASTA” (pronounced either “fast-uh” or “fast-A”; I tend to say the former but there is no difference.) format: The name of the sequence is prefaced by a greater-than sign <code>&gt;</code>, and the subsequent line(s) contain the associated sequence. The next sequence record begins with the next <code>&gt;</code>.</p>
<pre><code>&gt;moose
AGCTATAGAGCTA
&gt;camel
AACGATCGAGGTA</code></pre>
<p>All sequences are the same length (13 nucleotides) and presented as a multiple sequence alignment, so we can directly compare them. Looking column-by-column, there are 4 positions which differ between camel and moose and 9 that are the same. We would therefore say their <em>percent identity</em> is <span class="math inline">\(9/13\)</span> (69.2%), and the <em>percent difference</em> (or <em>proportion distance</em>) is <span class="math inline">\(4/13\)</span> (30.77%). We refer to the percent difference as a <strong>P-distance.</strong> We could also say that, considering these two sequences, 4 sites are <em>variable</em> and 9 sites are <em>conserved</em> (“constant”).</p>
<p>The problem with P-distance is that it’s calculated in a fairly naive manner if you are interested in the <strong>amount of evolutionary change.</strong> P-distance calculations make the assumption that each nucleotide difference is the result of a <em>single</em> substitution, and any unobserved differences are present because there were <em>no substitutions ever</em> at those sites. However, much like the honey badger, evolution does not care about simplicity. For example, consider position #1 where both species have the nucleotide <code>A</code>. The P-distance calculation assumes the most-recent common ancestor (MRCA) of camel and moose <em>also</em> had an <code>A</code> at this position, and nothing has changed since! This is quite a strong assumption, given evolution’s honey badger style. It is entirely possible that there WERE other changes that occured! Maybe the ancestor was a <code>T</code> and both camel and moose independently had a substitution to <code>T</code>. This would be an example of <em>convergent molecular evolution.</em> It is also entirely possible that the MRCA was an <code>A</code>, but along the evolutionary lineage to camels, the site substituted to <code>G</code>, then to <code>T</code>, and then back to <code>A</code> - this would be an example of <strong>multiple substitutions</strong> at a single site.</p>
<p>In fact, we are more interested in measures of <em>evolutionary <strong>divergence</strong></em>, which implies the total amount of evolutionary change, rather than simple distance calculations which only are able to consider <em>observed and present</em> nucleotide differences. We therefore desire a framework that can accommodate the serious possibility of complex and unobservable evolutionary histories. To do this, we use some fancier statistical models called <strong>Markov models.</strong> When we refer to “evolutionary” or “phylogenetic” models, this is what we mean!!</p>
</div>
<div id="markov-models" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Markov models</h2>
<p>Most simply, Markov models (or processes) are comprised of <em>probabilities (or rates) of instantaneous change among states</em>. A <em>realization</em> of a Markov process is called a <em>Markov chain.</em> We model nucleotide or protein evolution in this framework: An evolving site of DNA (aka a <em>column</em> in a multiple sequence alignment) is considered to be a Markov chain. …..<strong>Huh?</strong></p>
<p>Let’s break this down, one part at a time: What do we mean by <em>probabilities (or rates) of instantaneous change among states</em>? Working with nucleotides we have <em>four states</em>: A, C, G, and T! These are four things a DNA nucleotide can be. When a substitution occurs, nucleotides change into other nucleotides, and there is some <em>probability</em> a given nucleotide will change into another. In the simplest model, we would say there is an equal probability of any nucleotide changing to any other - the probability of mutating from <span class="math inline">\(A\rightarrow G\)</span> is the same as <span class="math inline">\(G\rightarrow T\)</span> is the same as <span class="math inline">\(T\rightarrow A\)</span>, and so on. We can write this out as a <em>matrix</em> called our <strong>instantaneous rate matrix</strong>, often referred to as the <strong><span class="math inline">\(Q\)</span> matrix</strong> since it is usually denoted with the letter <span class="math inline">\(Q\)</span>. Note that we do not consider substitutions to the same state, e.g. <span class="math inline">\(A\rightarrow A\)</span> is assumed to never happen, since it doesn’t make sense, and therefore the probability of this change is 0. Below, we use the <em>parameter</em> (fancy word for variable) <span class="math inline">\(\lambda\)</span> (Greek letter lambda) to represent the probability of each type of nucleotide substitution:</p>
<p>It’s not complete yet! There is another aspect to consider: In addition to just the probability of changing from one nucleotide to another, there is some <em>underlying probability</em> of BEING a given nucleotide in the first place! The matrix above assumes that all nucleotides are equally likely in a given gene, but this isn’t always fair to expect. We therefore also include separate <em>parameters</em> for the so-called “base frequencies”, also known as “equilibrium” or “stationary” frequencies<a href="#fn6" class="footnote-ref" id="fnref6"><sup>6</sup></a>. We represent these values with the Greek letter pi, <span class="math inline">\(\pi\)</span>.</p>
<p>Thus, the probability of a substitution from <code>A--&gt;T</code> is the <span class="math inline">\(\lambda\)</span>, the probability that a nucleotide changes, <em>weighted by</em> the probability <span class="math inline">\(\pi_T\)</span> of observing the <em>target</em> nucleotide T in the first place. <strong>This is your very first Markov model!!!</strong> It has five <em>parameters</em>:<span class="math inline">\(\lambda\)</span>, and the four values for <span class="math inline">\(\pi\)</span>, one for each nucleotide<a href="#fn7" class="footnote-ref" id="fnref7"><sup>7</sup></a>. This model, like virtually <em>all models</em> you will encounter in my lab, is therefore comprised of stationary frequencies and substitution probabilities (also referred to as substitution rates).</p>
<p>However, this is <em>still</em> pretty unrealistic: As biologists, we know that nucleotides aren’t all the same - there are purines (A and G) and pyrimidines (C and T). I will use <span class="math inline">\(R\)</span> to mean purine and <span class="math inline">\(Y\)</span> to mean pyrimidine. <span class="math inline">\(R\rightarrow R\)</span> and <span class="math inline">\(Y\rightarrow Y\)</span> mutations are known as <em>transitions</em>, and due to shared biochemistry, they occur much more easily than do <em>transversions</em> (<span class="math inline">\(R\rightarrow Y\)</span> or <span class="math inline">\(Y\rightarrow R\)</span>). We might want our model to reflect this biological information! We can do this by adding another parameter to our model that captures <strong>transition/tranversion bias</strong> - the underlying biological system’s preference for transitions over transversions. Commonly we represent this quantity with the Greek letter <span class="math inline">\(\kappa\)</span>, kappa. This therefore <em>more complex</em> than the simple one-rate model introduced above since it has another parameter. The more parameters, the more complex a model is.</p>
<p>Over the years, people have taken model complexity to its natural conclusion and have come up with a <em>lot</em> of ways to parameterize these models. The most complex model is referred to as the “GTR” (general time-reversible) model. In a GTR model, <em>all</em> substitution probabilities are different, and <em>all</em> states have their own stationary frequency. That said, the phrase “time-reversible” indicates that substitution probabilities are the same forwards and backwards in time: The probability of <span class="math inline">\(A\rightarrow G\)</span>, for example, is the same as <span class="math inline">\(G\rightarrow A\)</span>. GTR models are therefore comprised of a <em>symmetric</em>[Same forward and backwards!] matrix of substitution probabilities, and a <em>vector</em><a href="#fn8" class="footnote-ref" id="fnref8"><sup>8</sup></a> of state frequencies.</p>
<p><strong>The process of building a phylogeny using model-based methods is how these parameters get values!!</strong> TODO: software jointly <em>optimizes</em> (finds the best values for, using probability) the model’s parameters and the tree.</p>
<div id="key-properties-of-markov-processes" class="section level3" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Key properties of Markov processes</h3>
<p>Markov processes are <strong>memoryless</strong>: The probability of changing from one state to another depends ONLY on the current state, and NOT AT ALL on which states the chain was at previously. Consider this in terms of DNA: Imagine a site in the genome that has undergone substitutions over time as <span class="math inline">\(G \rightarrow A\rightarrow T \rightarrow C \rightarrow A \rightarrow T\)</span>. You’ll notice that, at two separate times in the DNA’s history, it experienced the <span class="math inline">\(A\rightarrow T\)</span> substitution. For the first one, the DNA site had previously been a <span class="math inline">\(G\)</span>, but for the second one, the DNA site had been a <span class="math inline">\(C\)</span>. The memoryless Markov process doesn’t care about the <span class="math inline">\(G\)</span> and the <span class="math inline">\(C\)</span> - the probability of going <span class="math inline">\(A\rightarrow T\)</span> does NOT depend on those previous nucleotides at all. The probability of going <span class="math inline">\(A\rightarrow T\)</span> is always the same probability forever and ever in this chain.</p>
<p>In addition, most applications (but this is starting to change, maybe!) in phylogenetics use <strong>time-reversible</strong> Markov models. Time-reversible implies that a given Markov chain has the same probabilities moving forwards and backwards in time<a href="#fn9" class="footnote-ref" id="fnref9"><sup>9</sup></a>, e.g. the probability of <span class="math inline">\(A\rightarrow G\)</span> is the same as <span class="math inline">\(G\rightarrow A\)</span>. This is not necessarily a fair biological assumption. The reason we tend to use time-reversible models is that they make the math and computing aspects of building and analyzing phylogenies, <em>a lot easier.</em> This is a common theme in modeling: Sometimes, the right modeling framework we’d like to use is simply too challenging for our computers. We need to use approximations and “short-cuts” a lot of the time to make sure the computer program can achieve results in, say, less than 100,000 years. The field is always moving forward, and new computational tricks are being established all the time! One day, I hope we will see the widespread use of <em>non</em>-time-reversible models in phylogenetics!</p>
</div>
<div id="markov-chains-and-multiple-sequence-alignments" class="section level3" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> Markov chains and multiple sequence alignments</h3>
<p>We consider each column in a multiple sequence alignment to be the <em>realization of a Markov process (aka a Markov chain) <strong>along</strong> a phylogeny</em>. The image below shows an example of the relationship between a column in an MSA and what it’s Markov chain <em>might</em> look like. What you see in the image below includes the <em>ancestral states</em> - ie, what <em>was</em> the nucleotide at that column for an unknown ancestor? In general, we can’t ever really know the true ancestors, since they are extinct. Our fancy statistical methods use <em>probability</em> to infer the most likely ancestral states at all internal nodes on the tree. We refer to this as <em>ancestral state reconstruction.</em> In the example below, I am assuming we know the ancestors with certainty, which would be the case for <em>simulated data</em> that I use a lot in my research.</p>
</div>
<div id="protein-models" class="section level3" number="5.2.3">
<h3><span class="header-section-number">5.2.3</span> Protein models</h3>
<p>Above, we described DNA models with four states (A, C, G, T). In fact, we also have models focused on <em>protein</em> or <em>codon</em> states too. Protein models are comprised of 20x20 matrices (since there are 20 amino acids), and codon models are comprised of 61x61 matrices (since there are 61 <em>sense</em> codons, and we ignore the 3 stop codons TAG, TAA, and TGA). We usually use protein models to build phylogenies out of amino-acid data. We do not usually use codon models to build phylogenies due to the computational intensiveness of dealing with 61x61 matrices, but we often use codon models to study <em>natural selection in DNA.</em> These models are introduced and described in the section <a href="measuring-evolutionary-rates.html#measuring-evolutionary-rates">Measuring evolutionary rates</a>.</p>
<p>Like nucleotide models, amino-acid models can be broken down into a symmetric 20x20 matrix and a vector of stationary frequencies. However, amino acids themselves <em>do not evolve</em> - nucleotides evolve, and this sometimes leads to a change in the protein product. Therefore, proteins evolve over time <em>due to changes in their coding DNA.</em> This is just something to bear in mind :)</p>
<p>Unfortunately, because protein models are 20x20, optimization is super hard. There are some statistical problems with trying to optimize values for <em>that many parameters</em> - you need a LOT of data to get precise, unbiased estimates. So, the field has decided to largely bail on this and instead use “pre-canned” models built from training datasets. Spielman thinks this is a mediocre prospect.</p>
</div>
<div id="rate-variation-or-heterogeneity" class="section level3" number="5.2.4">
<h3><span class="header-section-number">5.2.4</span> Rate variation or heterogeneity</h3>
<p>The models as described assume that all sites evolve according to the same process. Early on, people observed that different sites in DNA evolve at different rates. (example of protein image here). We want to include this information in our models! One common way to do this is with a <em>discrete gamma distribution</em>…Huh?</p>
<p>insert figure here, explain gamma.</p>
<p>In the end, you typically have <em>four</em> values drawn from this distribution to represent four different categories of sites. You actually multiply the model parameters by a given rate, and this increases or decreases the probabilities.</p>
</div>
</div>
<div id="choosing-a-model" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Choosing a model</h2>
<p>This is a broader concept in statistics: How do I choose a model for my data? What variables should I include, aka what level of complexity? We want to find the middle ground in under/overfitting and deal with the precision/bias tradeoff.</p>
<p>Remember the concept of <em>sample size</em>: In order to accurately infer values from data, our data should be large. Small samples result in bias. Therefore we’d like the find the model with the right number of parameters for the amount of data at hand.</p>
<p>Broadly speaking, there are two traditions of model selection: <em>relative model selection</em> and <em>absolute model selection.</em></p>
<p><strong>Relative model selection</strong> ranks all models by a given measurement that tells you how well your model works for the data at hand. In relative model selection, you provide a list of candidate models you could use for your dataset, a fancy software ranks the models in order, and you then select the best-ranking model for analyses. This approach has become the standard in phylogenetics because it is relatively simple to perform, <em>and</em> a highly usable software for doing this procedure was introduced in the late 1990. <strong>A significant chunk of my research program is trying to stop people from using this procedure!</strong> A lot of recent research, including my own, has demonstrated that for a lot of applications, relative model selection does not necessarily help you achieve a more correct answer, but it also doesn’t <em>harm</em> your analysis. It has just become a dogmatic assumption that everyone must use this analysis in molecular phylogenetics, and it is my solemn duty to tear it all down.</p>
<p>Importantly, relative model selection is entirely incapable of asking, “Should I even be considering these models <em>in the first place</em>?” It assumes any model you are ranking is a reasonable potential candidate model. Of course, we might prefer a procedure that actually asks if a given model is worth considering at all! This is <strong>absolute model selection,</strong> which are often referred to as <strong>model adequacy tests.</strong> This framework asks, “How well does <em>a given model</em> capture trends in my data?” It therefore only evaluates one model at a time, rather than comparing dozens or hundreds of candidate models to one another. This means absolute model selection is more computationally intensive (aka, takes a longer time), but in this researcher’s opinion, it is a <em>much</em> better bet than is relative model selection. <strong>Future research projects in my lab may seek to more formally examine the relationship between absolute and relative model selection: For example, are high-ranking models actually adequate? Are low-ranking models actually inadequate?</strong></p>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="6">
<li id="fn6"><p>There is actually <em>quite a bit more</em> math behind stationary frequencies, what they reall mean, and why we use them. We can discuss this separately if you really want the nitty gritty details, but the details aren’t strictly necessary for working with these models in my lab<a href="evolutionary-models.html#fnref6" class="footnote-back">↩︎</a></p></li>
<li id="fn7"><p>A true statistical phylogeneticist, however, would note there there are only <em>four</em> free parameters (aka parameters you need to understand the whole system): <span class="math inline">\(\lambda\)</span>, and only THREE values for <span class="math inline">\(\pi\)</span>! Since all the <span class="math inline">\(\pi\)</span>’s must sum to 1 (there is a 100% chance of being <em>any</em> nucleotide), you only need to know three of those values to know the fourth. Sneaky!<a href="evolutionary-models.html#fnref7" class="footnote-back">↩︎</a></p></li>
<li id="fn8"><p>Don’t panic! This just means “one-dimensional matrix” - a list!<a href="evolutionary-models.html#fnref8" class="footnote-back">↩︎</a></p></li>
<li id="fn9"><p>The actual definition is much more technical and beyond the scope of what you <em>need</em> to know for this research. If you want the real mathy details, let’s chat!<a href="evolutionary-models.html#fnref9" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mutiple-sequence-alignments.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="building-your-phylogeny.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
