<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Mutiple sequence alignments | An extra gentle introduction to molecular evolution</title>
  <meta name="description" content="4 Mutiple sequence alignments | An extra gentle introduction to molecular evolution" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Mutiple sequence alignments | An extra gentle introduction to molecular evolution" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="sjspielman/gentle_intro_molecular_evolution" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Mutiple sequence alignments | An extra gentle introduction to molecular evolution" />
  
  
  

<meta name="author" content="Stephanie J. Spielman" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="background-interpreting-building-and-using-phylogenies.html"/>
<link rel="next" href="evolutionary-models.html"/>
<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />












</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="background-molecular-evolution.html"><a href="background-molecular-evolution.html"><i class="fa fa-check"></i><b>2</b> Background: Molecular Evolution</a>
<ul>
<li class="chapter" data-level="2.1" data-path="background-molecular-evolution.html"><a href="background-molecular-evolution.html#evolution-in-populations"><i class="fa fa-check"></i><b>2.1</b> Evolution in populations</a></li>
<li class="chapter" data-level="2.2" data-path="background-molecular-evolution.html"><a href="background-molecular-evolution.html#evolution-in-dna"><i class="fa fa-check"></i><b>2.2</b> Evolution in DNA</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="background-molecular-evolution.html"><a href="background-molecular-evolution.html#a-brief-digression-snps"><i class="fa fa-check"></i><b>2.2.1</b> A brief digression: SNPs</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="background-molecular-evolution.html"><a href="background-molecular-evolution.html#fitness-consequences-of-mutations"><i class="fa fa-check"></i><b>2.3</b> Fitness consequences of mutations</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="background-interpreting-building-and-using-phylogenies.html"><a href="background-interpreting-building-and-using-phylogenies.html"><i class="fa fa-check"></i><b>3</b> Background: Interpreting, building, and using phylogenies</a>
<ul>
<li class="chapter" data-level="3.1" data-path="background-interpreting-building-and-using-phylogenies.html"><a href="background-interpreting-building-and-using-phylogenies.html#reading-and-interpreting-phylogenies"><i class="fa fa-check"></i><b>3.1</b> Reading and interpreting phylogenies</a></li>
<li class="chapter" data-level="3.2" data-path="background-interpreting-building-and-using-phylogenies.html"><a href="background-interpreting-building-and-using-phylogenies.html#the-pipeline-of-phylogenetic-reconstruction"><i class="fa fa-check"></i><b>3.2</b> The pipeline of phylogenetic reconstruction</a></li>
<li class="chapter" data-level="3.3" data-path="background-interpreting-building-and-using-phylogenies.html"><a href="background-interpreting-building-and-using-phylogenies.html#analyzing-sequences-in-a-phylogenetic-context"><i class="fa fa-check"></i><b>3.3</b> Analyzing sequences in a phylogenetic context</a></li>
<li class="chapter" data-level="3.4" data-path="background-interpreting-building-and-using-phylogenies.html"><a href="background-interpreting-building-and-using-phylogenies.html#gene-tree-vs-species-trees"><i class="fa fa-check"></i><b>3.4</b> Gene tree vs species trees</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="mutiple-sequence-alignments.html"><a href="mutiple-sequence-alignments.html"><i class="fa fa-check"></i><b>4</b> Mutiple sequence alignments</a></li>
<li class="chapter" data-level="5" data-path="evolutionary-models.html"><a href="evolutionary-models.html"><i class="fa fa-check"></i><b>5</b> Evolutionary models</a>
<ul>
<li class="chapter" data-level="5.1" data-path="evolutionary-models.html"><a href="evolutionary-models.html#background-why-use-evolutionary-models"><i class="fa fa-check"></i><b>5.1</b> Background: Why use evolutionary models?</a></li>
<li class="chapter" data-level="5.2" data-path="evolutionary-models.html"><a href="evolutionary-models.html#markov-models"><i class="fa fa-check"></i><b>5.2</b> Markov models</a></li>
<li class="chapter" data-level="5.3" data-path="evolutionary-models.html"><a href="evolutionary-models.html#flavors-dna-protein-codon-and-applications"><i class="fa fa-check"></i><b>5.3</b> Flavors (DNA, protein, codon) and applications</a></li>
<li class="chapter" data-level="5.4" data-path="evolutionary-models.html"><a href="evolutionary-models.html#phenomenological-vs-mechanistic"><i class="fa fa-check"></i><b>5.4</b> Phenomenological vs mechanistic</a></li>
<li class="chapter" data-level="5.5" data-path="evolutionary-models.html"><a href="evolutionary-models.html#choosing-a-model"><i class="fa fa-check"></i><b>5.5</b> Choosing a model</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="building-your-phylogeny.html"><a href="building-your-phylogeny.html"><i class="fa fa-check"></i><b>6</b> Building your phylogeny</a></li>
<li class="chapter" data-level="7" data-path="assessing-confidence-in-phylogenetic-inferences.html"><a href="assessing-confidence-in-phylogenetic-inferences.html"><i class="fa fa-check"></i><b>7</b> Assessing confidence in phylogenetic inferences</a>
<ul>
<li class="chapter" data-level="7.1" data-path="assessing-confidence-in-phylogenetic-inferences.html"><a href="assessing-confidence-in-phylogenetic-inferences.html#bootstrap-and-friends"><i class="fa fa-check"></i><b>7.1</b> Bootstrap and friends</a></li>
<li class="chapter" data-level="7.2" data-path="assessing-confidence-in-phylogenetic-inferences.html"><a href="assessing-confidence-in-phylogenetic-inferences.html#pp"><i class="fa fa-check"></i><b>7.2</b> PP</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="measuring-evolutionary-rates.html"><a href="measuring-evolutionary-rates.html"><i class="fa fa-check"></i><b>8</b> Measuring evolutionary rates</a>
<ul>
<li class="chapter" data-level="8.1" data-path="measuring-evolutionary-rates.html"><a href="measuring-evolutionary-rates.html#nucleotide-and-protein-relative-rates"><i class="fa fa-check"></i><b>8.1</b> Nucleotide and protein relative rates</a></li>
<li class="chapter" data-level="8.2" data-path="measuring-evolutionary-rates.html"><a href="measuring-evolutionary-rates.html#codon-dnds"><i class="fa fa-check"></i><b>8.2</b> Codon <span class="math inline">\(dN/dS\)</span></a></li>
<li class="chapter" data-level="8.3" data-path="measuring-evolutionary-rates.html"><a href="measuring-evolutionary-rates.html#mutation-selection-models"><i class="fa fa-check"></i><b>8.3</b> Mutation-selection models</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="the-molecular-clock.html"><a href="the-molecular-clock.html"><i class="fa fa-check"></i><b>9</b> The Molecular Clock</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">An extra gentle introduction to molecular evolution</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mutiple-sequence-alignments" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Mutiple sequence alignments</h1>
<p>Recall from the earlier section <a href="background-molecular-evolution.html#evolution-in-dna">Evolution in DNA</a>, that DNA mutations fall into two broad categories: a) substitutions (change in existing nucleotide(s)), and b) indels (nucleotides are inserted or deleted). In order to compare homologous sequences to one another, however, we need to line them up so they match. However, because of indels, two homologs are not necessarily going to have the same number of nucleotides. Some sequences will be shorter or longer, and sometimes <em>by a lot</em>.<a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a>.</p>
<p>We therefore need to create something called a <em>sequence alignment</em> (a “character matrix,” which you may recall from “Introduction to Evolution”, but for DNA and protein sequences instead of physical traits). When you align two sequences together, it’s called a “pairwise alignment.” When there are three or more sequences (pretty much all modern-day evolutionary studies), it’s called a <strong>multiple sequence alignment</strong>, often abbreviated as “MSA” and simply referred to as “alignments”<a href="#fn6" class="footnote-ref" id="fnref6"><sup>6</sup></a> There are many different software programs which will align your homologous sequences. In the end, sequences will have both nucleotides (or amino acids, depending on your dataset), and <strong>gaps</strong>, represented as dashes (“-”), which represent indel events. Each <strong>site</strong> is a <strong>column</strong> in the alignment.</p>
<p><strong>MSAs are <em>hard</em> to make. Really, really hard.</strong> So hard in fact, that we are pretty sure that even the fanciest most computer-y algorithms for building MSAs are error prone. The problem is, the software doesn’t know about the evolutionary process itself and instead relies on mathematical optimization to get the “right” alignment. As a consequence, we are pretty sure a lot of alignments are wrong, but we also don’t know <em>in what way</em> they are wrong. To make matters worse, MSAs are <strong>THE</strong> fundamental unit of data used in <em>all comparative sequence analyses and phylogenetics.</em></p>
<p>Because alignments are known to be error-prone, a lot of people (including myself) have tried think about ways we can quantify this uncertainty, or at least identify unreliable regions in a given MSA. Some people advocate for strategies like <em>trimming</em> or <em>filtering</em> alignments, where you remove parts of the alignment you infer to be problematic. Usually, these are regions that are really “gappy” (yes, we really use this word!). Indels are what makes aligning so tricky - indeed, if there were no indels and only substitutions, all homologous sequences would already be the same length and directly line up into columns in the first place. There has been conflicting evidence over the years whether filtering improves or worsens analysis outcomes. It doesn’t seem to hurt, but it also doesn’t seem to help very much.</p>
</div>
<div class="footnotes">
<hr />
<ol start="5">
<li id="fn5"><p>What’s <em>really</em> interesting though is that even though protein sequences and lengths change over evolutionary time, their overall <em>structure</em> really does not. Natural selection imposes strong constraints on protein evolution to maintain a particular structure, which in turn allows the protein to properly function<a href="mutiple-sequence-alignments.html#fnref5" class="footnote-back">↩︎</a></p></li>
<li id="fn6"><p>CAUTION: There is another type of “alignment” in bioinformatics that is used for genome sequencing. The type of alignment we are concerned with is those that compare <em>homologous sequences from DIFFERENT individuals/species</em>.<a href="mutiple-sequence-alignments.html#fnref6" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="background-interpreting-building-and-using-phylogenies.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="evolutionary-models.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
